src/components/app/SourceFetcher.js

```diff
@@ -40,6 +40,8 @@ type DispatchProps = {|
 type Props = ConnectedProps<{||}, StateProps, DispatchProps>;
 
 class SourceFetcherImpl extends React.PureComponent<Props> {
+  _archiveCache: Map<string, Promise<Uint8Array>> = new Map();
```

Comment:
Could the fact that we keep all uncompressed downloaded sources without ever cleaning them be a problem?

I don't know really if the archives we download are big, especially uncompressed. Maybe in the future we should store them in IDB rather than keeping them in memory? Also this would make it possible to keep them between profiler instances?

Reply:
The ones I've seen are usually a few hundred KB, and almost all are less than 10MB. So I don't expect a lot of trouble from this. But caching across profiler instances can still make sense.
I'm just not too excited about adding more IndexedDB usage, especially thinking about tests... But I can file a follow-up bug about it.